### DISCUSSION TITLE
Local webhook with self signed certificates

### DISCUSSION THREAD

**Discussion by AristeoMontoya** [firefly-iii/firefly-iii#7509] [Firefly III] [answered]
Hi, I've been trying to write a simple webhook when a transaction is created, that webhook listener (a simple Spring service with a few endpoints) is currently running on the same device but firefly doesn't seem to trust my self signed certificate according to this message in the logs:
```
curl error 60: ssl certificate problem: self signed certificate
```
I've tried mounting the .pem file as suggested in #5996 but I'm still getting that log message, does anyone know how can I get around this?

**Comment by JC5** [comment]
Does the PEM file have the entire chain?

**Reply by AristeoMontoya** [reply]
Not entirely sure, I used keytool to generate a p12 file like so
```
keytool -genkeypair -alias firefly -keyalg RSA -keysize 4096 -storetype PKCS12 -keystore firefly.p12 -validity 3650
```

And then I converted that to a pem file using the following command:
```
openssl pkcs12 -in firefly.p12 -out firefly.pem -nodes
```

I'm new to working with certs, so I'm not entirely sure

**Reply by JC5** [reply, marked as answer]
You could also just use the `http` connection?

**Reply by AristeoMontoya** [reply]
I just confirmed that it is possible through http, for some reason I was completely sure that I've read that it wasn't possible in the documentation. This is solved since https is not a requirement for me (it runs locally without being exposed). Thanks
